homeassistant/components/bring/notify.py

```diff
@@ -0,0 +1,112 @@
+"""Notify platform for the Bring! integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from bring_api.exceptions import BringRequestException
+from bring_api.types import BringNotificationType
+import voluptuous as vol
+
+from homeassistant.components.notify import NotifyEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import config_validation as cv, entity_platform
+from homeassistant.helpers.config_validation import make_entity_service_schema
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN
+from .const import ATTR_ITEM_NAME, ATTR_NOTIFICATION_TYPE, MANUFACTURER, SERVICE_NAME
+from .coordinator import BringData, BringDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Bring notify entity platform."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        BringNotify(coordinator, bring_list=bring_list, entry=config_entry)
+        for bring_list in coordinator.data.values()
+    )
+
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        "send_message",
```

Comment:
We don't allow overriding a builtin service. That would be confusing to users. If there's no message to send the integration shouldn't implement a notify platform.

Reply:
But the service is not overriden 🤔 Instead an additional service with a custom service schema is registered alongside the builtin service from the notify platform. This is currently required, as the notify entity service does not allow extra data keys like the legacy notify platform (required for item field). I also changed notification_type to message, as that is basically the message that is sent, just that only 4 predefined keywords are allowed. In contrast to other notification services, bring does not have a free text input for the message. The custom service schema does therefore define the message field as radio input.

I know the new notification entity component is still a work in progress, @joostlek asked me if i could implement this feature with it. Maybe this can serve as some kind of use case for the further development @jbouwh ^^